### ISSUE TITLE
[BUG] SfSearchBar invalid html issues

### ISSUE THREAD

**Issue by DavidGolodetsky** [vuestorefront/storefront-ui#2103]
**Describe the bug**
Nuxt HTML-validator throws three errors on the SfSearchBar component:
1. `<input>` element does not have a `<label>`
2. `<button>` is missing required "type" attribute
3. `<button>` must have accessible text

**How to reproduce**
You can try to put the output of the component to the [W3C validator](https://validator.w3.org/#validate_by_input+with_options)

**Expected behavior**
The SfSearchBar component restructured in a way that HTML would be valid

**Actual behavior**
The input tag doesn't have a label, although it's pronounced via screen reader. but the search button is pronounced as a "button", would be nice to add an a11y label there.

**Screenshots**
![image](https://user-images.githubusercontent.com/32278933/141166210-962a5b4f-a1f2-4591-b683-49636327616f.png)

**Additional context**
Related to #2097

**Comment by justyna-13** [comment]
@DavidGolodetsky do you have the same errors in W3C validator? I've checked SfSearchBar and I can't see them. There is one error regarding a wrong id (it's already reported and will be fixed) but no errors regarding a button and an input. When it comes to button tag - its type attribute isn't required and there is an icon inside a button so without text but it has an aria-label so that assistive technology will be able to read what it is for

**Comment by DavidGolodetsky** [comment]
huh, indeed I don't see these errors in W3C as well. I thought they should be the same. 
1. Button `type` attribute isn't required, but it's considered good practice to always provide it to be more descriptive and if there is no `type` specified it will act as a `submit` and it'll try to reload the page. I was thinking of creating a separate ticket for that issue to be handled in the SfButton component.
2. input has an aria-label but not the search button. I believe it should have an aria-label as well, otherwise, a user might take it for "reset" button or something else

**Comment by justyna-13** [comment]
I think that this validator on nuxt is more strict and shows errors even if they aren't considered errors by W3C. 
We'll discuss the button type attribute in our team. Regarding aria-label - it's added on SfButton in our search component- so far it's in English but if you would like to change it you can use slot "icon" and have your custom button :)

**Comment by DavidGolodetsky** [comment]
Great, then please take a look at my comment about the button type attribute [here]. If you'll decide that we don't need to change anything I guess this issue can be closed.(https://github.com/vuestorefront/storefront-ui/issues/2102)
